### PR TITLE
standard dockerfile

### PR DESCRIPTION
**Pull request by efiShtain**
removed old docker resources (build directory)
added standard docker files (created using new docker init command)

### CODE REVIEW

File: compose.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# This service is built from the Dockerfile in the current directory.
+# You can add other services your application may depend on here, such as a
+# database or a cache. For examples, see the Awesome Compose repository:
+# https://github.com/docker/awesome-compose
+services:
+  server:
+    build:
+      context: .
+      target: final
+    command: ["up", "--mode=dev"]
```
Comment (daabr) [inline compose.yaml:15]: Why dev mode? Even if experimental, the cloud instance should be the "full, real AK". More importantly, dev mode lacks data persistence and durability, which are critical for a non-local AK instance.

This includes: a Temporal server, DB(s) for Temporal and AK, and a Redis server for AK - all these components need to be independent and separate from day 1.

File: compose.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    ports:
+      - 9980:9980
+# The commented out section below is an example of how to define a PostgreSQL
```
Comment (daabr) [inline compose.yaml:18]: "The commented out section below" - if it's not commented-out, why include it in this PR?

Also, we haven't decided yet on which DB to use, where, and how - this is scheduled for Wednesday, but if needed & possible we can discuss tomorrow too.

Lastly, all cloud configuration orchestration (i.e. meant to be used by us, not users) should be maintained in a non-public repo due to their sensitive nature. Or am I misunderstanding the purpose of this PR?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+# Build the application.
+# Leverage a cache mount to /go/pkg/mod/ to speed up subsequent builds.
+# Leverage a bind mount to the current directory to avoid having to copy the
+# source code into the container.
+RUN --mount=type=cache,target=/go/pkg/mod/ \
+    --mount=type=bind,target=. \
+    CGO_ENABLED=0 go build -o /bin/ak ./cmd/ak
+
+################################################################################
```
Comment (itayd) [inline Dockerfile:26]: remove example comments

File: compose.yaml (hunk 1 above)
Comment (efiShtain) [reply compose.yaml:15]: this is for developers who want to just start it and work with it, 
this is not a production compose

I will add more features to the compose file later to support a database and persistence temporal

File: compose.yaml (hunk 2 above)
Comment (efiShtain) [reply compose.yaml:18]: all cloud related stuff are in infrastructure repo, 
except for the dockerfile and this compose file which are standard in the main repo

**Comment by itayd** [comment]
@daabr to approve

**Review by daabr — APPROVED** [review]

**Review by itayd — APPROVED** [review]

**Pull request merged**